presto-main/src/main/java/com/facebook/presto/operator/scalar/SequenceFunction.java

```diff
@@ -17,28 +17,88 @@
 import com.facebook.presto.spi.block.Block;
 import com.facebook.presto.spi.block.BlockBuilder;
 import com.facebook.presto.spi.block.BlockBuilderStatus;
+import com.facebook.presto.spi.type.AbstractFixedWidthType;
+import com.facebook.presto.spi.type.StandardTypes;
 import com.facebook.presto.type.SqlType;
 import com.google.common.primitives.Ints;
+import org.joda.time.DateTime;
+import org.joda.time.Months;
 
 import static com.facebook.presto.spi.StandardErrorCode.INVALID_FUNCTION_ARGUMENT;
 import static com.facebook.presto.spi.type.BigintType.BIGINT;
+import static com.facebook.presto.spi.type.TimestampType.TIMESTAMP;
 import static com.facebook.presto.util.Failures.checkCondition;
 
 public final class SequenceFunction
 {
     private SequenceFunction() {}
 
+    private static Block fixedWidthSequence(long start, long stop, long step, AbstractFixedWidthType type)
+    {
+        checkCondition(step != 0, INVALID_FUNCTION_ARGUMENT, "step must not be zero");
+        checkCondition(step > 0 ? stop >= start : stop < start, INVALID_FUNCTION_ARGUMENT,
+                "sequence stop value should be greater than or equal to start value if step is greater than zero otherwise stop should be less than start");
+
+        int length = Ints.checkedCast((stop - start) / step + 1L);
+        checkCondition(length > 0, INVALID_FUNCTION_ARGUMENT, "length of sequence is 0");
```

Comment:
On second inspection, this can never be less than 1 since `(stop - start) / step > 0` for every integer value of `start`, `stop`, and `step`. With that in mind, do we even need this assertion?